src/sys/windows/selector.rs

```diff
@@ -122,7 +127,12 @@ impl SockState {
                  * events that the user is interested in. Therefore, cancel the pending
                  * poll operation; when we receive it's completion package, a new poll
                  * operation will be submitted with the correct event mask. */
-                self.cancel()?;
+                let result = self.cancel();
```

Comment:
I haven't been able to find any documentation on `NtCancelIoFileEx` (the system call made here), do you have a documentation @dtacalau?

I want to know what kind of errors can be returned here because maybe we can ignore some of them.

Reply:
@Thomasdezeeuw not much, seems it can return any NTSTATUS. The only I was able to found: 
- some of these unit tests check result of `NtCancelIoFileEx` https://git.reactos.org/?p=reactos.git;a=blob;hb=6d0c07c44fae0a93c57f8d3410545e0f3c201287;f=modules/rostests/winetests/ntdll/pipe.c#l384
- Win list of NTSTATUS codes https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55

Reply:
Always great when you're developing against an API for which you don't have docs, right?

Reply:
rhetorical question I guess :)